controllers/frontend_controller_suite_test.go

```diff
@@ -921,4 +923,545 @@ var _ = ginkgo.Describe("Dependencies", func() {
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "cloud.redhat.com/v1",
+						Kind:       "Frontend",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      FrontendName,
+						Namespace: FrontendNamespace,
+					},
+					Spec: crd.FrontendSpec{
+						EnvName:        FrontendEnvName,
+						Title:          "",
+						DeploymentRepo: "",
+						Frontend: crd.FrontendInfo{
+							Paths: []string{"/things/test"},
+						},
+						Image: "my-image:version",
+						Module: &crd.FedModule{
+							ManifestLocation: "/apps/inventory/fed-mods.json",
+							Modules:          []crd.Module{},
+						},
+						SearchEntries: []*crd.SearchEntry{{
+							ID:          "test",
+							Href:        "/test/href",
+							Title:       "Test",
+							Description: "Test description",
+						}, {
+							ID:          "test2",
+							Href:        "/test2/href",
+							Title:       "Test2",
+							Description: "Test2 description",
+						}},
+					},
+				}
+				gomega.Expect(k8sClient.Create(ctx, frontend)).Should(gomega.Succeed())
+				frontendEnvironment := &crd.FrontendEnvironment{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "cloud.redhat.com/v1",
+						Kind:       "FrontendEnvironment",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      FrontendEnvName,
+						Namespace: FrontendNamespace,
+					},
+					Spec: crd.FrontendEnvironmentSpec{
+						SSO:      "https://something-auth",
+						Hostname: "something",
+						Monitoring: &crd.MonitoringConfig{
+							Mode: "app-interface",
+						},
+						GenerateNavJSON: false,
+					},
+				}
+				gomega.Expect(k8sClient.Create(ctx, frontendEnvironment)).Should(gomega.Succeed())
+				createdConfigMap := &v1.ConfigMap{}
+				gomega.Eventually(func() bool {
+					err := k8sClient.Get(ctx, configMapLookupKey, createdConfigMap)
+					if err != nil {
+						return err == nil
+					}
+					if len(createdConfigMap.Data) != 2 {
+						return false
+					}
+					return true
+				}, timeout, interval).Should(gomega.BeTrue())
+				gomega.Expect(createdConfigMap.Name).Should(gomega.Equal(FrontendEnvName))
+				gomega.Expect(createdConfigMap.Data).Should(gomega.Equal(map[string]string{
+					"fed-modules.json":  "{\"testSearchIndex\":{\"manifestLocation\":\"/apps/inventory/fed-mods.json\",\"fullProfile\":false}}",
+					"search-index.json": "[{\"id\":\"test-search-index-test-search-index-env-test\",\"href\":\"/test/href\",\"title\":\"Test\",\"description\":\"Test description\"},{\"id\":\"test-search-index-test-search-index-env-test2\",\"href\":\"/test2/href\",\"title\":\"Test2\",\"description\":\"Test2 description\"}]",
+				}))
+				gomega.Expect(createdConfigMap.ObjectMeta.OwnerReferences[0].Name).Should(gomega.Equal(FrontendEnvName))
+			})
+
+			ginkgo.By("from multiple Frontend resources", func() {
+				ctx := context.Background()
+
+				configMapLookupKey := types.NamespacedName{Name: FrontendEnvName2, Namespace: FrontendNamespace}
+
+				frontend2 := &crd.Frontend{
```

Comment:
@Hyperkid123 a bunch of this is duplicated with the `frontend` variable in the first test... Is there a way to de-dupe some of the common values and focus on the ones that actually change in the test itself?

Reply:
You need a unique variable otherwise it will mess with the values. I tried to make it a bit more "generic" in the last two test suites. I might do the same for the search index.